packages/app/src/cli/services/context/identifiers-extensions.ts

```diff
@@ -131,18 +141,54 @@ export async function deployConfirmed(
 
   // For extensions we also need the match by ID, not only UUID (doesn't apply to functions)
   for (const [localIdentifier, uuid] of Object.entries(validMatches)) {
-    const registration = extensionRegistrations.find((registration) => registration.uuid === uuid)
+    const registration = uuidUidStrategyExtensions.find((registration) => registration.uuid === uuid)
     if (registration) validMatchesById[localIdentifier] = registration.id
   }
 
   return {
     extensions: validMatches,
     extensionIds: {...validMatchesById, ...extensionsIdsNonUuidManaged},
-    extensionsNonUuidManaged,
+    extensionsNonUuidManaged: {...extensionsNonUuidManaged},
   }
 }
 
-async function ensureNonUuidManagedExtensionsIds(
+function matchWebhooks(remoteSource: RemoteSource, extension: ExtensionInstance) {
+  const remoteVersionConfig = remoteSource.activeVersion?.config ?? remoteSource.draftVersion?.config
+  const remoteVersionConfigObj = remoteVersionConfig ? JSON.parse(remoteVersionConfig) : undefined
+  const localConfig = extension.configuration as unknown as SingleWebhookSubscriptionType
+  return remoteVersionConfigObj.topic === localConfig.topic && remoteVersionConfigObj.uri === localConfig.uri
+}
+
+async function loadExtensionIds(
+  remoteConfigurationRegistrations: RemoteSource[],
+  developerPlatformClient: DeveloperPlatformClient,
+  localExtensionRegistrations: ExtensionInstance[],
+  extensionsToCreate: LocalSource[],
+  validMatches: {[key: string]: unknown},
+  validMatchesById: {[key: string]: unknown},
+) {
+  localExtensionRegistrations.forEach((local) => {
+    const possibleMatches = remoteConfigurationRegistrations.filter((remote) => {
+      return remote.type === developerPlatformClient.toExtensionGraphQLType(local.graphQLType)
+    })
+
+    let match: RemoteSource | undefined
+    if (local.isSingleStrategyExtension && possibleMatches.length === 1) {
+      match = possibleMatches[0]
+    } else if (local.isDynamicStrategyExtension) {
+      match = possibleMatches.find((possibleMatch) => matchWebhooks(possibleMatch, local))
+    }
```

Comment:
This is also something very webhooks specific, could this also be abstracted away to the generic interface? This logic will also move to the server eventually though right? Not a blocker

Reply:
The idea was to not modify the generic interface at all and keep all the webhooks stuff isolated so that we can easily remove it in the future.

Reply:
makes sense 👍